tests/test_subdomains.py

```diff
@@ -275,16 +275,11 @@ def test_multi_sets(self):
         Ny = Nx
         n_domains = 2
 
-        n = Dimension(name='n')
```

Comment:
@rhodrin as promised, these can go...

Reply:
What if want to use same dimension for two subdomains?

Reply:
you mean two `MultiSubDomains`?
well, they _must_ be different by construction right? fusing iteration over two distinct MultiSubDomains might create all sort of issues (eg swapping thicknesses)